src/Wyam.Modules.AmazonWebServices/GenerateCloudSearchData.cs

```diff
@@ -0,0 +1,111 @@
+        /// <param name="idMetaKey">The meta key represeting the unique ID for this document.  If NULL, the Document.Id will be used.</param>
+        /// <param name="bodyField">The field name for the document contents.  If NULL, the document contents will not be written to the data.</param>
+        public GenerateCloudSearchData(string idMetaKey, string bodyField)
+        {
+            _idMetaKey = idMetaKey;
+            _bodyField = bodyField;
+            _fieldMap = new Dictionary<string, string>();
+        }
+
+        /// <summary>
+        /// Adds a mapping from meta key to CloudSearch field. When provided, the contents of the meta key will be written to the profided field name.
+        /// </summary>
+        /// <param name="fieldName">The CloudSearch field name.</param>
+        /// <param name="metaKey">The meta key.</param>
+        public GenerateCloudSearchData MapMetaField(string fieldName, string metaKey)
+        {
+            if (fieldName == null)
+            {
+                throw new ArgumentNullException(nameof(fieldName));
+            }
+            if (metaKey == null)
+            {
+                throw new ArgumentNullException(nameof(metaKey));
+            }
+            _fieldMap.Add(fieldName, metaKey);
+            return this;
+        }
+
+
+        public IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs, IExecutionContext context)
+        {
+            var sb = new StringBuilder();
+            var sw = new StringWriter(sb);
+
+            using (JsonWriter writer = new JsonTextWriter(sw))
+            {
+                writer.WriteStartArray();
+
+                foreach (var doc in inputs)
+                {
+                    writer.WriteStartObject();
+
+                        writer.WritePropertyName("type");
+                        writer.WriteValue("add");
+
+                        writer.WritePropertyName("id");
+                        writer.WriteValue(
+                            _idMetaKey != null ? doc.String(_idMetaKey) : doc.Id
+                            );
+                                          
+                        writer.WritePropertyName("fields");
+                        writer.WriteStartObject();
+
+                            if (_bodyField != null)
+                            {
+                                writer.WritePropertyName(_bodyField);
+                                writer.WriteValue(doc.Content);
+                            }
+
+                            foreach(var field in _fieldMap)
+                            {
+                                writer.WritePropertyName(field.Key);
+                                writer.WriteValue(doc.String(field.Value));
+                            }
+
+                        writer.WriteEndObject();
+
+                    writer.WriteEndObject();
+                }
+
+                writer.WriteEndArray();
+
+                return new[] { context.GetNewDocument(sw.ToString()) };
```

Comment:
The `GetNewDocument()` call is changing after I merge the stuff I'm currently working on, but I'll take care of it on my branch - just a heads up.